nixos/modules/services/games/minecraft-server.nix

```diff
@@ -158,6 +158,32 @@ in {
           + "-XX:MinHeapFreeRatio=5 -XX:MaxHeapFreeRatio=10";
         description = "JVM options for the Minecraft server.";
       };
+
+      fifo = mkOption {
+        type = types.nullOr types.str;
+        default = null;
+        example = "/run/mincraft/stdin";
+        description = "create a FiFo device to send commands to the minecraft process";
+      };
+
+      warn.stopMessage = mkOption {
+        type = types.listOf types.str;
+        default = [ "The server is going to stop" "3" "2" "1" ];
+        description = ''
+          Send a stop warning to users on the server, when stoping the systemd service.
+          Only ha an effect when <option>services.minecraft-server.fifo</option> is not
+          <literal>null</literal>.
```

Comment:
With the fifo not being optional, this should probably become optional now, defaulting to not being enabled.

I suggest redoing these options like this though: Have a single option called `stopCommands` of type `types.nullOr types.lines`, which represents a bash script whose stdout is piped to the Fifo. By default it would be `null`, and the example could be
```nix
''
  echo "say This server is stopping in 3.."
  sleep 1
  echo "say 2.."
  sleep 1
  echo "say 1.."
  sleep 1
''
```

With this, the functionality can be expressed with a single option, while it can then also be used to do more complicated shutdown behaviors.